hydromt/raster.py

```diff
@@ -2247,6 +2248,245 @@ def tile_window(shape, px):
+                lr = t_r.transform(bounds[2], bounds[1])
+                out_bounds = [ul[0], lr[1], lr[0], ul[1]]
+
+                temp_bounds = [
+                    max(out_bounds[0], obj_bounds[0]),
+                    max(out_bounds[1], obj_bounds[1]),
+                    min(out_bounds[2], obj_bounds[2]),
+                    min(out_bounds[3], obj_bounds[3]),
+                ]
+
+                _window = [
+                    abs(x - y) / obj_res for x, y in zip(temp_bounds, obj_bounds)
+                ]
+
+                u = math.floor(round(_window[3], 2))
+                l = math.floor(round(_window[0], 2))
+                h = obj_shape[0] - u - math.floor(round(_window[1], 2))
+                w = obj_shape[1] - l - math.floor(round(_window[2], 2))
+
+                temp = obj[u : u + h, l : l + w]
+
+                tile = np.empty((px_size, px_size), np.float64)
+                tile[:] = nodata
+
+                temp_transform = Affine(
+                    obj_res,
+                    0,
+                    obj_bounds[0] + l * obj_res,
+                    0,
+                    -obj_res,
+                    obj_bounds[3] - u * obj_res,
+                )
+
+                rasterio.warp.reproject(
+                    temp.values,
+                    tile,
+                    src_transform=temp_transform,
+                    src_crs=obj.raster.crs,
+                    dst_transform=transform,
+                    dst_crs="EPSG:3857",
+                    dst_nodata=nodata,
+                    resampling=resampling,
+                )
+
+                temp = xr.DataArray(
+                    name=temp.name,
+                    dims=("y", "x"),
+                    coords={
+                        "y": np.arange(
+                            bounds[3] + 0.5 * transform[4], bounds[1], transform[4]
+                        )[:px_size],
+                        "x": np.arange(
+                            bounds[0] + 0.5 * transform[0],
+                            bounds[2] + 0.5 * transform[0],
+                            transform[0],
+                        )[:px_size],
+                    },
+                    data=tile,
+                )
+
+                del tile
+
+                temp.raster.set_crs(3857)
+
+                temp.raster.set_nodata(nodata)
+
+                if driver == "netcdf4":
+                    path = join(ssd, f"{row}.nc")
+                    temp = temp.raster.gdal_compliant()
+                    temp.to_netcdf(path, engine="netcdf4", **kwargs)
+                elif driver in gis_utils.GDAL_EXT_CODE_MAP:
+                    ext = gis_utils.GDAL_EXT_CODE_MAP.get(driver)
+                    path = join(ssd, f"{row}.{ext}")
+                    temp.raster.to_raster(path, driver=driver, **kwargs)
+                else:
+                    raise ValueError(f"Unkown file driver {driver}")
+                file.write(f"{path}\n")
```

Comment:
If this is the only time you write to file, it might be more efficient to just open up the filew with a context manager, write to it, and close it again. especially if you're concerned about memory.

Reply:
It's an inhouse function, which itself calls writing functions/ a context manager. Can't put one here,